__docs__/wonder-blocks-switch/switch.stories.tsx

```diff
@@ -0,0 +1,167 @@
+                ariaLabel="test switch"
+                checked={checkedTwo}
+                onChange={setCheckedTwo}
+                icon={<Icon icon={icons.search} />}
+            />
+        </View>
+    );
+};
+
+Controlled.play = async ({canvasElement}) => {
+    const canvas = within(canvasElement);
+
+    const switchWithIcon = canvas.getByTestId("test-switch");
+    const switchInput = canvas.getByRole("switch", {name: "test switch"});
+
+    await userEvent.tab();
+    await userEvent.tab();
+
+    expect(switchWithIcon).toHaveStyle(
+        "background-color: rgba(33, 36, 44, 0.5)",
+    );
+    expect(switchWithIcon).toHaveStyle("outline: 2px solid rgb(24, 101, 242)");
+    expect(switchInput).toHaveAttribute("aria-checked", "false");
+
+    await userEvent.click(switchWithIcon);
+
+    expect(switchWithIcon).toHaveStyle("background-color: rgb(24, 101, 242)");
+    expect(switchInput).toHaveAttribute("aria-checked", "true");
+};
+
+Controlled.parameters = {
+    docs: {
+        description: {
+            story: `The switch is a controlled component, so state should be used to keep track of
+            whether it is checked or not. The \`onChange\` prop is optional in case the toggle
+            will be wrapped in a larger clickable component.`,
+        },
+    },
+};
+
+export const Disabled: StoryComponentType = () => (
+    <View style={styles.column}>
+        <Switch checked={false} disabled={true} />
+        <Strut size={Spacing.xSmall_8} />
+        <Switch checked={true} disabled={true} />
+        <Strut size={Spacing.xSmall_8} />
+        <Switch
+            checked={false}
+            disabled={true}
+            icon={<Icon icon={icons.search} />}
+        />
+        <Strut size={Spacing.xSmall_8} />
+        <Switch
+            checked={true}
+            disabled={true}
+            icon={<Icon icon={icons.search} />}
+        />
+    </View>
+);
+
+Disabled.parameters = {
+    docs: {
+        description: {
+            story: `The switch can be disabled.`,
+        },
+    },
+};
+
+export const WithIcon: StoryComponentType = () => {
+    return (
+        <View style={styles.column}>
+            <Switch checked={false} icon={<Icon icon={icons.search} />} />
+            <Strut size={Spacing.xSmall_8} />
+            <Switch checked={true} icon={<Icon icon={icons.search} />} />
+        </View>
+    );
+};
+
+WithIcon.parameters = {
+    docs: {
+        description: {
+            story: `The \`medium\` size switch can take an \`icon\` prop, which will be rendered inside the slider.`,
+        },
+    },
+};
+
+export const InsideCell: StoryComponentType = () => {
+    const [checked, setChecked] = React.useState(false);
+
+    return (
+        <CompactCell
+            title="Switch as a right accessory"
+            onClick={() => {
+                setChecked(!checked);
+            }}
+            rightAccessory={<Switch checked={checked} />}
+        />
+    );
+};
```

Comment:
Hmmmmmmmmm... This story still isn't behaving the way I'd expect with a screen reader. When you press on the cell, it doesn't read out the new checked value. I would expect it to work similarly to option items in a single select or multiselect dropdown.

Let me know if you'd like help investigating this!

Reply:
update from dicussions offline: we decided the whole cell should _not_ be clickable, and it is best to pass in html `label` tag elements so that clicking them also toggles the switch. this way, the screen reader only focuses on the switch and the click area is also expanded to include the `label` element.